src/time.rs

```diff
@@ -169,6 +169,39 @@ impl GpsTime {
     pub fn floor_to_epoch(&self, soln_freq: f64) -> GpsTime {
         GpsTime(unsafe { c_bindings::floor_to_epoch(self.c_ptr(), soln_freq) })
     }
+
+    pub fn to_gal(self) -> GalTime {
+        GalTime {
+            wn: self.wn() - c_bindings::GAL_WEEK_TO_GPS_WEEK as i16,
+            tow: self.tow(),
+        }
+    }
+
+    pub fn to_bds(self) -> BdsTime {
+        let bds = GpsTime::new_unchecked(
+            self.wn() - c_bindings::BDS_WEEK_TO_GPS_WEEK as i16,
```

Comment:
Probably a bit of an extreme edge case but is it possible for `self.wn() - c_bindings::BDS_WEEK_TO_GPS_WEEK as i16` to be <= 0? Looks like it's defined as `#define BDS_WEEK_TO_GPS_WEEK 1356` so

```
(base) {10:16}~/projects/swiftnav-rs:jbangelo/gal-bds-glo-time ✗ ➭ junco gps2utc 1356 0
2005-12-31T23:59:47Z
```
If for some reason you're working with data that old it's possible to underflow? I guess beidou didn't exist at the point so the conversion is kind of meaningless anyway.

Reply:
But maybe worth an `assert!(self.wn() - c_bindings::BDS_WEEK_TO_GPS_WEEK as i16 > 0)`?

Reply:
Good idea. Initially I wanted to allow conversion of any valid GPS time to the equivalent GAL/BDS/GLO time, even if that meant to week numbers ended up being negative. In reality I don't think that's going to a realistic use case, you shouldn't be running into timestamps before the defined beginning of the time system. I'll add some asserts and change the sanity checks in the `new()` functions to only allow week numbers `>= 0`.

Reply:
I've added several asserts to make sure that the times are valid in the destination time base, and made clear in the doc comments that they will assert in certain situations.